diff -ur a/internal/libraw_x3f.cpp b/internal/libraw_x3f.cpp
--- a/internal/libraw_x3f.cpp	2015-11-23 23:43:29.000000000 -0800
+++ b/internal/libraw_x3f.cpp	2016-08-13 22:12:52.064303000 -0700
@@ -1,4 +1,4 @@
-﻿/* Library for accessing X3F Files 
+/* Library for accessing X3F Files 
 ----------------------------------------------------------------
 BSD-style License
 ----------------------------------------------------------------
@@ -297,7 +297,7 @@
 } x3f_camf_t;
 
 typedef struct x3f_directory_entry_header_s {
-  uint32_t identifier;        /* Should be ´SECp´, "SECi", ... */
+  uint32_t identifier;        /* Should be SECp, "SECi", ... */
   uint32_t version;           /* 0x00020001 is version 2.1  */
   union {
     x3f_property_list_t property_list;
@@ -318,7 +318,7 @@
 } x3f_directory_entry_t;
 
 typedef struct x3f_directory_section_s {
-  uint32_t identifier;          /* Should be ´SECd´ */
+  uint32_t identifier;          /* Should be SECd */
   uint32_t version;             /* 0x00020001 is version 2.1  */
 
   /* 2.0 Fields */
@@ -328,7 +328,7 @@
 
 typedef struct x3f_header_s {
   /* 2.0 Fields */
-  uint32_t identifier;          /* Should be ´FOVb´ */
+  uint32_t identifier;          /* Should be FOVb */
   uint32_t version;             /* 0x00020001 means 2.1 */
   uint8_t unique_identifier[SIZE_UNIQUE_IDENTIFIER];
   uint32_t mark_bits;
diff -ur a/libraw/libraw_const.h b/libraw/libraw_const.h
--- a/libraw/libraw_const.h	2015-11-23 23:43:29.000000000 -0800
+++ b/libraw/libraw_const.h	2016-08-13 22:12:20.126377100 -0700
@@ -88,7 +88,7 @@
 enum LibRaw_decoder_flags
 {
 /*  LIBRAW_DECODER_LEGACY = 1, */
-/*  LIBRAW_DECODER_FLATFIELD = 1<<1, */
+    LIBRAW_DECODER_FLATFIELD = 1<<1,
     LIBRAW_DECODER_USEBAYER2 = 1<<3,
     LIBRAW_DECODER_HASCURVE = 1<<4,
     LIBRAW_DECODER_SONYARW2 = 1<<5,
